client/login/wp-login/index.jsx

```diff
@@ -104,12 +118,16 @@ class Login extends React.Component {
 				</a>;
 		}
 
-		const goBackLink = ! magicLoginView && <a
-			href="#"
-			key="back-link"
-			onClick={ this.goBack }>
-				<Gridicon icon="arrow-left" size={ 18 } /> { this.props.translate( 'Return' ) }
-			</a>;
+		let goBackLink;
+		if ( this.state.loaded && window.history.length > 1 ) {
```

Comment:
Humm. I wonder why React throws this warning since the browser history should be empty on page load, meaning the return link wouldn't be displayed in that case.